truss/contexts/image_builder/serving_image_builder.py

```diff
@@ -339,6 +339,35 @@ def copy_into_build_dir(from_path: Path, path_in_build_dir: str):
         # Copy over truss
         copy_tree_path(truss_dir, build_dir, ignore_patterns=truss_ignore_patterns)
 
+        if config.docker_server is not None:
```

Comment:
This code section is getting too long, can we organize things in helper functions?

Reply:
will move it out as a new test file